Extension/src/Debugger/extension.ts

```diff
@@ -28,7 +30,67 @@ export function initialize(context: vscode.ExtensionContext): void {
     if (os.platform() === 'win32') {
         disposables.push(vscode.debug.registerDebugConfigurationProvider('cppvsdbg', new CppVsDbgConfigurationProvider(configurationProvider)));
     }
-    disposables.push(vscode.debug.registerDebugConfigurationProvider('cppdbg', new CppDbgConfigurationProvider(configurationProvider)));
+    const provider: CppDbgConfigurationProvider = new CppDbgConfigurationProvider(configurationProvider);
+    disposables.push(vscode.debug.registerDebugConfigurationProvider('cppdbg', new QuickPickConfigurationProvider(provider)));
+
+    disposables.push(vscode.commands.registerTextEditorCommand("C_Cpp.BuildAndDebugActiveFile", async (textEditor: vscode.TextEditor, edit: vscode.TextEditorEdit, ...args: any[]) => {
+        const folder: vscode.WorkspaceFolder = vscode.workspace.getWorkspaceFolder(textEditor.document.uri);
+        if (!folder) {
+            // Not enabled because we do not react to single-file mode correctly yet.
+            // We get an ENOENT when the user's c_cpp_properties.json is attempted to be parsed.
+            // The DefaultClient will also have its configuration accessed, but since it doesn't exist it errors out.
+            vscode.window.showErrorMessage('This command is not yet available for single-file mode.');
+            return Promise.resolve();
+        }
+
+        if (!util.fileIsCOrCppSource(textEditor.document.uri.fsPath)) {
+            vscode.window.showErrorMessage('The command is disabled because the active file is not a C or C++ source file.');
+            return Promise.resolve();
+        }
+        let configs: vscode.DebugConfiguration[] = (await provider.provideDebugConfigurations(folder)).filter(config => {
+            return config.name.indexOf("Build and Debug active file") !== -1;
+        });
+
+        interface MenuItem extends vscode.QuickPickItem {
+            configuration: vscode.DebugConfiguration;
+        }
+
+        const items: MenuItem[] = configs.map<MenuItem>(config => {
+            return {label: config.name, configuration: config};
+        });
+
+        vscode.window.showQuickPick(items, {placeHolder: "Select a configuration"}).then(async selection => {
```

Comment:
if `items` is empty, we should not show the quick pick.  (On Windows, we might want to show a message that msvc is not supported yet)

Reply:
I show "No compiler found" in the quick pick.